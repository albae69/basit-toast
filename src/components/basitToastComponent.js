import React, { useEffect } from "react";
import { SuccessIcon, WarningIcon, ErrorsIcon, DefaultIcon } from "../icons";
import { SUCCESS, DEFAULT, WARNING, ERRORS } from "../type/index";
import PropTypes from "prop-types";

export const BasitToast = ({
  isOpen,
  severity = "default",
  position = "bottomcenter",
  message,
  handleClose,
  handleCloseDuration = 3000,
}) => {
  useEffect(() => {
    if (isOpen) {
      setTimeout(handleClose, handleCloseDuration);
    }
  }, [isOpen, handleClose]);

  return (
    <div
      className={`basit-toast basit-toast__${severity} basit-toast__${position} ${
        isOpen ? "show" : ""
      }`}
    >
      {severity === DEFAULT && <DefaultIcon />}
      {severity === SUCCESS && <SuccessIcon />}
      {severity === WARNING && <WarningIcon />}
      {severity === ERRORS && <ErrorsIcon />}

      <p className="basit-toast-title">
        {message.length > 25 ? `${message.substr(0, 25)}...` : message}
      </p>
    </div>
  );
};

BasitToast.propTypes = {
  /**
   * The boolean for visible toast
   */
  isOpen: PropTypes.bool.isRequired,
  /**
   * Severity type for toast
   */
  severity: PropTypes.oneOf(["default", "success", "warning", "errors", "info"])
    .isRequired,
  /**
   * position toast if show
   */
  position: PropTypes.oneOf([
    "topright",
    "topleft",
    "topcenter",
    "bottomright",
    "bottomleft",
    "bottomcenter",
  ]).isRequired,
  /**
   * message in toast
   */
  message: PropTypes.string.isRequired,

  /**
   * handleClose
   */
  handleClose: PropTypes.func.isRequired,

  /**
   * handleCloseDuration time
   */
  handleCloseDuration: PropTypes.number,
};
