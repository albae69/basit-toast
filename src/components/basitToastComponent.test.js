import React from "react";
import renderer from "react-test-renderer";
import { BasitToast } from "./basitToastComponent";

test("BasitToast have been test", () => {
  const components = renderer.create(
    <BasitToast
      isOpen={true}
      position="bottomcenter"
      severity="success"
      message="Hello"
      handleClose={() => console.log("closed")}
      handleCloseDuration={3000}
    />
  );

  let tree = components.toJSON();
  tree.props.isOpen;
  tree.props.position;
  tree.props.severity;
  tree.props.message;
  tree.props.handleClose;
  tree.props.handleCloseDuration;
});
