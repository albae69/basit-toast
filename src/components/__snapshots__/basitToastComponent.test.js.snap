// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasitToast have been test 1`] = `
<div
  className="basit-toast basit-toast__success basit-toast__success show"
>
  <svg
    fill="none"
    height="25"
    viewBox="0 0 25 25"
    width="25"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.5 0C5.60757 0 0 5.60757 0 12.5C0 19.3924 5.60757 25 12.5 25C19.3924 25 25 19.3924 25 12.5C25 5.60757 19.3924 0 12.5 0ZM10.2163 18.7728L5.3726 13.39L6.80228 12.1034L10.1677 15.8425L18.149 6.33774L19.6238 7.57211L10.2163 18.7728Z"
      fill="white"
    />
  </svg>
  <p
    className="basit-toast-title"
  >
    Hello
  </p>
</div>
`;
